packages/core/src/lib/utils/index.ts

```diff
@@ -1,7 +1,295 @@
+    },
+  };
+
+  static tx = {
+    serialize(tx: Transaction): Uint8Array {
+      return Utils.serialize(tx);
+    },
+    deserialize(tx: Uint8Array): Transaction {
+      return Utils.deserialize(tx);
+    },
+    serializeData(
+      txData: InitDid | InitSessionKey,
+      category: TxCategory
+    ): Uint8Array {
+      return Utils.serializeData(txData, category);
+    },
+    deserializeData(
+      txData: Uint8Array,
+      category: TxCategory
+    ): InitDid | InitSessionKey {
+      return Utils.deserializeData(txData, category);
+    },
+  };
+
+  /**
+   * @param chainAgnosticAddress in CAIP10 - example: eip155:1:0xabc...
+   */
+  private static toUniversal(chainAgnosticAddress: string): UniversalAccount {
+    const [chain, chainId, address] = chainAgnosticAddress.split(':');
+
+    if (chain.toLocaleLowerCase() === 'eip155') {
+      if (chainId === '1') {
+        return {
+          chain: CHAIN.ETHEREUM,
+          chainId: CHAIN_ID.ETHEREUM.MAINNET,
+          account: address,
+        };
+      } else if (chainId === '11155111') {
+        return {
+          chain: CHAIN.ETHEREUM,
+          chainId: CHAIN_ID.ETHEREUM.SEPOLIA,
+          account: address,
+        };
+      } else
+        return {
+          chain: CHAIN.ETHEREUM,
+          chainId,
+          account: address,
+        };
+    } else if (chain.toLocaleLowerCase() === 'push') {
+      if (chainId.toLocaleLowerCase() === 'mainnet') {
+        return {
+          chain: CHAIN.PUSH,
+          chainId: CHAIN_ID.PUSH.MAINNET,
+          account: address,
+        };
+      } else if (chainId.toLocaleLowerCase() === 'devnet') {
+        return {
+          chain: CHAIN.PUSH,
+          chainId: CHAIN_ID.PUSH.DEVNET,
+          account: address,
+        };
+      } else
+        return {
+          chain: CHAIN.PUSH,
+          chainId,
+          account: address,
+        };
+    } else if (chain.toLocaleLowerCase() === 'solana') {
+      if (chainId.toLocaleLowerCase() === '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp') {
+        return {
+          chain: CHAIN.SOLANA,
+          chainId: CHAIN_ID.SOLANA.MAINNET,
+          account: address,
+        };
+      } else if (
+        chainId.toLocaleLowerCase() === 'EtWTRABZaYq6iMfeYKouRu166VU2xqa1'
+      ) {
+        return {
+          chain: CHAIN.SOLANA,
+          chainId: CHAIN_ID.SOLANA.DEVNET,
+          account: address,
+        };
+      } else if (
+        chainId.toLocaleLowerCase() === '4uhcVJyU9pJkvQyS88uRDiswHXSCkY3z'
+      ) {
+        return {
+          chain: CHAIN.SOLANA,
+          chainId: CHAIN_ID.SOLANA.TESTNET,
+          account: address,
+        };
+      } else
+        return {
+          chain: CHAIN.SOLANA,
+          chainId: chainId,
+          account: address,
+        };
+    } else
+      return {
+        chain: chain,
+        chainId: chainId,
+        account: address,
+      };
+  }
+
+  private static toChainAgnostic(universalAccount: UniversalAccount): string {
```

Comment:
We can have a "CHAIN" : "CAIP NETWORK" mapping  in config for this rather than using else if 

"ETHEREUM" : "eip155"
"SOLANA" : "solana" 

etc